lightly_studio/docs/docs/index.md

```diff
@@ -806,3 +806,208 @@ with open("export.txt", "w") as f:
+
+In this example we create an auto-labeling operator plugin powered by LightlyTrain, so make sure `lightly-train` is installed via `pip install lightly-train`. Compared to the Hello World example, this operator plugin introduces two inputs: the model name and the confidence threshold used for predictions. These parameters let you choose a pre-trained LightlyTrain model and control how confident detections must be before they are written back to LightlyStudio.
+
+```python title="lightly_train_auto_label_od_operator.py"
+from dataclasses import dataclass
+
+import lightly_train
+from PIL import Image
+from lightly_train._commands.predict_task_helpers import prepare_coco_entries as prepare_entries
+
+from lightly_studio.models.annotation.annotation_base import AnnotationCreate
+from lightly_studio.models.annotation_label import AnnotationLabelCreate
+from lightly_studio.plugins.base_operator import BaseOperator, OperatorResult
+from lightly_studio.plugins.parameter import FloatParameter, StringParameter
+from lightly_studio.resolvers import annotation_label_resolver, annotation_resolver, image_resolver
+
+
+def _preload_label_map(session, class_names):
+    """Pre-creates all necessary labels in the DB and returns a lookup map.
+
+    Args:
+        session: Database session.
+        class_names: List of class names the model supports (e.g. ['car', 'person']).
+
+    Returns:
+        A dictionary mapping label names to their DB UUIDs.
+    """
+    label_map = {}
+
+    for name in class_names:
+        # Check if label exists in db
+        label = annotation_label_resolver.get_by_label_name(session=session, label_name=name)
+
+        # Create if missing
+        if label is None:
+            label_create = AnnotationLabelCreate(annotation_label_name=name)
+            label = annotation_label_resolver.create(session=session, label=label_create)
+
+        label_map[name] = label.annotation_label_id
+
+    return label_map
+
+@dataclass
+class LightlyTrainAutoLabelingODOperator(BaseOperator):
+    name: str = "LightlyTrain: OD auto-labeling"
+    description: str = "This plugin allows to use pre-trained LightlyTrain models to perform auto-labeling for Object Detection."
+
+    @property
+    def parameters(self):
+        return [
+            StringParameter(
+                name="Model",
+                required=True,
+                description="The name of the pre-trained model to be used.",
+                default="dinov3/convnext-tiny-ltdetr-coco"
+            ),
+            FloatParameter(
+                name="Threshold",
+                default=0.4,
+                description="The confidence threshold to be applied to the predictions."
+            ),
+        ]
+
+    def execute(self, *, session, dataset_id, parameters):
+        try:
+            model = lightly_train.load_model(parameters["Model"])
+        except ValueError:
+            return OperatorResult(success=False, message="model_name is invalid.")
```

Comment:
This is a risky, as there might be other `ValueError`s that are caught and then the error message is misleading. Best rather copy the error message from e of  `except ValueError as e` .

https://github.com/lightly-ai/lightly-train/blob/bb247bfc9989648a070e3a57dd9186d2018252a2/src/lightly_train/_task_models/task_model_helpers.py#L229
```suggestion
        try:
            model = lightly_train.load_model(parameters["Model"])
        except ValueError as e:
            return OperatorResult(success=False, message=f"Model load failed: {str(e)})
```